src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -155,6 +156,22 @@ public override HttpManagementPayload CreateHttpManagementPayload(string instanc
             return await this.GetDurableOrchestrationStatusAsync(state, showHistory, showHistoryOutput);
         }
 
+        /// <inheritdoc />
+        public override async Task<IList<DurableOrchestrationStatus>> GetStatusAsync(bool showHistory = false, bool showHistoryOutput = false, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // TODO this cast is to avoid to change DurableTask.Core. Change it to use TaskHubClient.
+            var serviceClient = (DurableTask.AzureStorage.AzureStorageOrchestrationService)this.client.serviceClient;
+            var states = await serviceClient.GetOrchestrationStateAsync(cancellationToken);
+            var tasks = new List<Task<DurableOrchestrationStatus>>();
```

Comment:
By default, the `Task<T>` class will create a list size of some small number, maybe 4 or 11, I forget. If we add more items than this, `Task<T>` will silently discard its internal buffer, create a new one of double size, and copy all the items from the first buffer to the new buffer. It will keep doing this until we finish adding elements to the list. This is very inefficient and can cause performance problems with the .NET Garbage Collector. The best practice is to use the `List<T>(int)` constructor overload to specify an initial list capacity that meets our needs.  In this case, I think you should say the following:

```csharp
var result = new List<DurableOrchestrationStatus>(states.Count);
```

Reply:
Oh, no. I didn't know that. That is why people use List<T>(int) ... I double check the spec.  However, this code has gone. :) Now no problem.